keywords.robot

```diff
@@ -702,45 +702,37 @@ Get DUT To Start State
     END
 
 Turn On Power Supply
-    ${pc}=    Get Variable Value    ${POWER_CTRL}
-    IF    'sonoff' == '${pc}'
-        ${state}=    Sonoff On
-    ELSE
-        ${state}=    Rte Relay Set    on
-    END
+    Rte Psu On
 
 Power Cycle On
-    [Documentation]    Clears telnet buffer and perform full power cycle with
-    ...    RTE relay set to ON.
-    [Arguments]    ${power_button}=${FALSE}
+    [Documentation]    Clears telnet buffer and performs full power cycle
+    ...    by setting power supply to OFF, and then to ON. If platform needs
+    ...    additional power button press, it will be used as well, so at the
+    ...    end of this keyword platform starts booting. This is controlled via
+    ...    the DEFAULT_POWER_STATE_AFTER_FAIL variable defined in platform config.
+
+    Variable Should Exist    ${DEFAULT_POWER_STATE_AFTER_FAIL}
     IF    "${OPTIONS_LIB}"=="dcu" and "${POWER_CTRL}"=="none"
         Execute Reboot Command
         Sleep    5s
         RETURN
     END
+
     Restore Initial DUT Connection Method
     ${pc}=    Get Variable Value    ${POWER_CTRL}
-    IF    'sonoff' == '${pc}'
-        Sonoff Power Cycle On
-    ELSE IF    'obmcutil' == '${pc}'
+    IF    'obmcutil' == '${pc}'
         OBMC Power Cycle On
     ELSE
-        Rte Relay Power Cycle On
-    END
-
-    IF    ${power_button}    Rte Power On
-
-Rte Relay Power Cycle On
-    [Documentation]    Clears telnet buffer and perform full power cycle with
-    ...    RTE relay set to ON.
-    Rte Relay Set    off
-    Sleep    10
-    TRY
-        Telnet.Read
-    EXCEPT
-        Log    Could not clear Telnet buffer
+        Rte Psu Off
+        Sleep    10
+        TRY
+            Telnet.Read
+        EXCEPT
+            Log    Could not clear Telnet buffer
+        END
```

Comment:
if it is a generic keyword, is it actually good to use Telnet for every possible platform?

I know it was here before, but... Shouldn't it use `Read From Terminal` or whatever the generic keyword is called?

Reply:
I think that was the assumption back then, yes.

This might be still valid, as telnet would be the only one we can use that early (we won't use ssh here).

Nevertheless, updated that, still works after this change on mentioned platforms.